pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/AvoidFieldNameMatchingMethodNameRule.java

```diff
@@ -28,29 +31,36 @@ public Object visit(ASTClassOrInterfaceDeclaration node, Object data) {
     }
 
     @Override
-    public Object visit(ASTClassOrInterfaceBody node, Object data) {
-        int n = node.getNumChildren();
-        List<ASTFieldDeclaration> fields = new ArrayList<>();
-        Set<String> methodNames = new HashSet<>();
-        for (int i = 0; i < n; i++) {
-            Node child = node.getChild(i);
-            if (child.getNumChildren() == 0) {
-                continue;
-            }
-            child = child.getChild(child.getNumChildren() - 1);
-            if (child instanceof ASTFieldDeclaration) {
-                fields.add((ASTFieldDeclaration) child);
-            } else if (child instanceof ASTMethodDeclaration) {
-                methodNames.add(((ASTMethodDeclaration) child).getName().toLowerCase(Locale.ROOT));
+    public Object visit(ASTEnumDeclaration node, Object data) {
+        return super.visit(node, data);
+    }
+
+    @Override
+    public Object visit(JavaNode node, Object data) {
+        if (node instanceof ASTClassOrInterfaceBody || node instanceof ASTEnumBody) {
```

Comment:
This is a very uncommon pattern. Usually we do this in such a case:

```java
public Object visit(ASTClassOrInterfaceBody node, Object data) {
    handleClassOrEnum(node, data);
    return super.visit(node, data);
}

public Object visit(ASTEnumBody node, Object data) {
    handleClassOrEnum(node, data);
    return super.visit(node, data);
}

private void handleClassOrEnum(JavaNode node, Object data) {
    // impl here
}
```

Reply:
Thank you for your instruction. I will rewrite this part!